### DISCUSSION TITLE
能不能添加对IPv6的支持！

### DISCUSSION THREAD

**Discussion by hpwater** [kkkgo/PaoPaoGateWay#10] [想法]
生成的Openwrt镜像只支持IPv4，不支持IPv6，IPv6地址在出国还是由好处的，不容易被做掉。

**Comment by kkkgo** [comment]
如果节点入口本身是v4的，但是节点出口具备v6的话，那么访问v6的网站不会有问题，因为会直接转发原始请求到节点；
如果是想连接v6地址的节点的话，是不支持的，理论上可以支持，但在构建的时候就已经去掉了v6，原因有：
- 减少镜像体积
- v6和v4双栈存在的时候，v6会被优先使用，这会导致产生一系列的不稳定因素，如域名解析、订阅下载可能会失败或者需要重试多次，禁用v6可以提高稳定性和处理速度
- v6直接暴露在公网，服务存在安全隐患，虽然目前的配置看起来是足够安全的，但不能保证上游程序是否存在漏洞
- v6目前家宽是动态IP，而无论是DHCPv6还是SLAAC，都存在IP过期的问题，当IP过期失效而又没使用新的有效IP的期间内，系统会拥有一个无法使用的v6地址，这就会导致双栈的时候系统优先使用”无法使用的v6地址“去联网，不仅是无法连接v6节点的问题，这也产生了第二点的问题
- 如果需要连接纯v6地址的节点，你可以在其他系统或者docker上使用IPv6，配置运行你的服务，然后开一个socks5端口给PaoPaoGateway即可，使用`mode=socks5`模式，可以满足你其他拓展需求。
- 以后有时间会考虑兼容v6和测试（提供定制选项的形式），但目前主要在更新PaoPaoDNS，PaoPaoGateway以实现稳定工作为主

**Reply by fsiisx** [reply]
以后有时间会考虑兼容v6和测试（提供定制选项的形式）

请问大佬这个有计划了吗？

**Reply by kkkgo** [reply]
相关功能已更新，查看最新release：https://github.com/kkkgo/PaoPaoGateWay/releases/
